include/ensmallen_bits/adam/adam_impl.hpp

```diff
@@ -37,7 +37,7 @@ AdamType<UpdateRule>::AdamType(
               tolerance,
               shuffle,
               UpdateRule(epsilon, beta1, beta2),
-              NoDecay(),
+              DecayPolicyType(),
```

Comment:
Perhaps, we should add the option to pass an instantiation of the update and decay policy, see:

https://github.com/mlpack/ensmallen/blob/a83ff61cbd42d5ea9b86ebdbedff53d73740d33c/include/ensmallen_bits/sgd/sgd.hpp#L96-L97

Reply:
Since all the existing `UpdateRule`s take `epsilon`, `beta1`, and `beta2`, I think it's fine to leave it as is.  But the new `AdamW` update also has the `weightDecay` parameter.  Therefore I would actually suggest a new `AdamW` class that wraps `Adam<AdamWUpdate>` and allows a parameter for `weightDecay` in the constructor.

And if `DecayPolicyType` is not necessary at all (and I don't think it is) then we don't need to worry about that part.